### PR TITLE
Actions: Compare proofs with master

### PR DESCRIPTION
**Pull request by kcalvinalvin**
Fetches the master branch and diffs the proof against the one generated by the PR.

It's currently breaking because the master binary doesn't have the flags I'm using in the script. Works after #215 is merged.

### CODE REVIEW

File: test/csn_bridge.sh [hunk 1]
```diff
@@ -10,7 +10,8 @@ set -Eeuo pipefail
 
 GENPROOFS="$1"
 IBDSIM="$2"
-TEST_DATA=$(mktemp -d)
+TEST_DATA="$3"
```
Comment (dergoegge) [inline test/csn_bridge.sh:13]: Could still assign a temp dir if $3 is not given.

File: .github/workflows/go.yml [hunk 2]
```diff
@@ -54,4 +60,63 @@ jobs:
...
-        ./test/csn_bridge.sh "./cmd/utreexoserver/utreexoserver" "./cmd/utreexoclient/utreexoclient"
+        ./test/csn_bridge.sh "./cmd/utreexoserver/utreexoserver" "./cmd/utreexoclient/utreexoclient" "${{ github.workspace }}/testdatapath"
+
+    - name: Upload testdata
+      uses: actions/upload-artifact@v2
+      with:
+        name: testdatapath
+        path: "${{ github.workspace }}/testdatapath"
+
+  check-proof-backwards:
```
Comment (dergoegge) [inline .github/workflows/go.yml:71]: It looks a bit weird that there are so many duplicate steps, like "setup env", "Install Go", "Get deps" and "Build" are the same as in the previous job. 
Could this not just be a couple extra steps in the first job?
After "Run integration tests" add: "Checkout master", "Build", "Check backwards".

File: .github/workflows/go.yml [hunk 3]
```diff
@@ -54,4 +60,63 @@ jobs:
...
+        cd $GOPATH/src/github.com/${{ github.repository }}
+        go get -v -d ./...
+
+    - name: Build
+      shell: bash
+      run: |
+        cd $GOPATH/src/github.com/${{ github.repository }}/cmd/utreexoserver
+        go install -v
+
+    - name: checkout
```
Comment (dergoegge) [inline .github/workflows/go.yml:112]: Is this only needed because `./test/check_proofs_backwards.sh` does not yet exist on master?

Comment (kcalvinalvin) [reply .github/workflows/go.yml:112]: Yes. Not exactly the prettiest... I guess we could update this afterwards

File: test/csn_bridge.sh (hunk 1 above)
Comment (kcalvinalvin) [reply test/csn_bridge.sh:13]: Ok would be helpful for testing locally. I'll add this

File: .github/workflows/go.yml (hunk 2 above)
Comment (kcalvinalvin) [reply .github/workflows/go.yml:71]: Jobs don't share state so they would have to be uploaded to github artifacts.

I think it'd be more maintainable/functional if the second job fetches and builds itself. That process is fairly quick as well

File: test/csn_bridge.sh (hunk 1 above)
Comment (kcalvinalvin) [reply test/csn_bridge.sh]: Added

**Pull request merged**